admin/tabs/roles/editor/form/RoleForm.ts

```diff
@@ -0,0 +1,132 @@
+import {RoleFormModel} from '@xh/hoist/admin/tabs/roles/editor/form/RoleFormModel';
+import {RoleMemberType} from '@xh/hoist/admin/tabs/roles/HoistRole';
+import {warningBanner} from '@xh/hoist/admin/tabs/roles/warning/WarningBanner';
+import {form} from '@xh/hoist/cmp/form';
+import {grid} from '@xh/hoist/cmp/grid';
+import {hbox, hframe, vbox, vframe} from '@xh/hoist/cmp/layout';
+import {storeFilterField} from '@xh/hoist/cmp/store';
+import {hoistCmp, HoistProps, uses, XH} from '@xh/hoist/core';
+import {formField} from '@xh/hoist/desktop/cmp/form';
+import {select, textArea, textInput} from '@xh/hoist/desktop/cmp/input';
+import './RoleForm.scss';
+import {panel} from '@xh/hoist/desktop/cmp/panel';
+import {recordActionBar} from '@xh/hoist/desktop/cmp/record';
+import {toolbar} from '@xh/hoist/desktop/cmp/toolbar';
+import {capitalizeWords} from '@xh/hoist/format';
+import {Icon} from '@xh/hoist/icon';
+
+export const roleForm = hoistCmp.factory({
+    className: 'xh-admin-role-form',
+    displayName: 'RoleForm',
+    model: uses(RoleFormModel),
+    render({className, model}) {
+        return vframe({
+            className,
+            items: [
+                vbox({
+                    className: `${className}__fields`,
+                    item: form({
+                        fieldDefaults: {
+                            commitOnChange: true,
+                            minimal: true
+                        },
+                        items: [
+                            hbox(
+                                formField({
+                                    disabled: model.isEditingExistingRole,
+                                    field: 'name',
+                                    flex: 1,
+                                    item: textInput({autoFocus: true})
+                                }),
+                                formField({
```

Comment:
Should we make category required?  If we want to keep optional, let's spec enableClear. Note that no-category roles end up in a slightly wonky "(Blanks)" group in the main grid

Reply:
Don't think it should be required, but good point about the wonky blanks group...